src/request_queue.js

```diff
@@ -40,7 +40,13 @@ const queuesCache = new LruCache({ maxLength: MAX_OPENED_QUEUES }); // Open queu
  * @ignore
  */
 const validateAddRequestParams = (request, opts) => {
-    checkParamPrototypeOrThrow(request, 'request', Request, 'Apify.Request');
+    let newRequest;
+    try {
```

Comment:
Thank you for PR. This is a bit error-prone as for example if you pass a string as `request` parameter it also tries to create an object from it.

We should try this only for objects whose prototype is not `Apify.Request`. Ie.

```javascript
checkParamOrThrow(request, 'request', 'Object'); // We need object for both cases and throw an error otherwise

if (!(request instanceof Request)) request = new Request(request);
```


```

Reply:
@mtrunkat, got it.

I chose to defer validation of the `request` parameter to the `Request` constructor as I noticed it had a whole bunch of other `checkParamOrThrow()` calls which individually validate a `Request` object constructor. So if a `string` was passed, it'd choke on the constructor—although the error returned is more opaque (i.e. `Parameter "opts" of type Object must be provided`).